tabbycat/api/serializers.py

```diff
@@ -1344,3 +1348,17 @@ class Meta:
     class Meta:
         model = Team
         fields = ('team', 'rounds')
+
+
+class ActionLogSerializer(serializers.ModelSerializer):
+
+    id = serializers.CharField(read_only=True)
+    user = serializers.CharField(read_only=True)
+    ip_address = serializers.IPAddressField(read_only=True)
+    timestamp = serializers.DateTimeField(read_only=True)
+    type = serializers.ChoiceField(choices=ActionLogEntry.ACTION_TYPE_CHOICES, read_only=True)
+    object_id = serializers.CharField(read_only=True)
```

Comment:
Would these be necessary? They are model fields included by being on line 1364. They shouldn't be considered read-only in the serializer itself.

There should be a (this time read-only) URL field for the action-log object.

The real problem to solve for #1993 is to create a polymorphic URL field (called `object` or `target`) that uses `type` and `object_id` to create an API URL to the affected object.

Reply:
Sure enough, I have a solution to this figured out. My only problem is what do you think is the best way to handle actionlog entries on objects that don't have corresponding API endpoints (like checkins or preformed panels?

Reply:
Fair point, but I don't think that's a problem; the API is quite comprehensive (including with how to checkin an object or preformed panels). For checkins, I think we use the checkin status as a state on the corresponding object, so you'd link to the ballot rather than ballot/checkin.